enumerable.rb

```diff
@@ -0,0 +1,179 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    count = 0
+    while count < arr.length
+      yield(arr[count])
+      count += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    count = 0
+    while count < arr.length
+      yield(arr[count], count)
+      count += 1
+    end
+    arr
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each { |it| new_arr << it if yield(it) }
+    new_arr
+  end
+
+  def my_all?(param = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+      return true
+    elsif param.nil?
+      my_each { |num| return false if num.nil? || num == false }
+    elsif !param.nil? && (param.is_a? Class)
+      my_each { |num| return false if num.class != param }
+    elsif !param.nil? && param.instance_of?(Regexp)
+      my_each { |_num| return false unless arg.match(n) }
+    else
+      my_each { |_num| return false if n != param }
+    end
+    true
+  end
+
+  def my_any?(param = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+      return true
+    elsif param.nil?
+      my_each { |num| return true if num.nil? || num == true }
+    elsif !param.nil? && (param.is_a? Class)
+      my_each { |num| return true if num.instance_of?(param) }
+    elsif !param.nil? && param.instance_of?(Regexp)
+      my_each { |_num| return true unless arg.match(num) }
+    else
+      my_each { |_num| return true if num != param }
+    end
+    true
+  end
+
+  def my_none?(param = nil)
+    if !block_given? && param.nil?
+      my_each { |num| return true if num }
+      return false
+    end
+
+    if !block_given? && !param.nil
+
+      if param.is_a?(Class)
+        my_each { |num| return false if num.instance_of?(param) }
+        return true
+      end
+
+      if param.instance_of?(Regexp)
+        my_each { |num| return false if arg.match(num) }
+        return true
+      end
+
+      my_each { |num| return false if num == param }
+      true
+    end
+    my_any? { |it| return false if yield(it) }
+    true
+  end
+
+  def my_count(num = nil)
+    arr = instance_of?(Array) ? self : to_a
+    return arr.length unless block_given? || num
+
+    return arr.my_select { |item| item == num }.length if num
+
+    arr.my_select { |item| yield(item) }.length
+  end
+
+  def my_map(par_ = nil)
+    new_arr = []
+    if par_
+      each { |i| new_arr << par_.call(i) }
+    else
+      each { |i| new_arr << yield(i) }
+      new_arr
+    end
+  end
+
+  def my_inject(number = nil, _syn = nil)
```

Comment:
enumerables #my_inject raises a "LocalJumpError" when no block or argument is given
```
      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
```


enumerables #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method
```
      Failure/Error: expect(array.my_inject(20, :*)).to eq array.inject(20, :*)
```

  
enumerables #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range
```
      Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)
```
# 
### `my_inject` should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)).